gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -548,6 +548,15 @@ public SubscriptionEntity process(ProcessSubscriptionEntity processSubscription,
                 subscription.setClosedAt(new Date());
             }
 
+            if (
```

Comment:
In this graviteeVersion, ApiKeyService.generateForSubscription do this check properly, and is called by :
- ApiKeyService.generate
- ApiKeyService.renew

So, instead of adding this if block, maybe we could just move up the `apiKeyService.generate` call which is already present at line 627 ?

Reply:
Done